agent/api/task.go

```diff
@@ -97,6 +97,15 @@ type Task struct {
 	KnownStatusTimeUnsafe time.Time `json:"KnownTime"`
 	knownStatusTimeLock   sync.RWMutex
 
+	// PullStartedAt is the timestamp when the task start pulling
+	PullStartedAt time.Time `json:"PullStartedAt"`
+	// PullStoppedAt is the timestamp when the task finished pulling
+	PullStoppedAt time.Time `json:"PullStoppedAt"`
```

Comment:
since we're using pointers to the `time.Time` fields in the Get* methods - would it make sense have these fields use pointers as well? 

then we can do nil checks (which i think would be clearer) instead of the `task.PullStartedAt == (time.Time{})` and also we'd avoid going back and forth between using values in some places and the pointers in others. 

also the marshaled zero value for `time.Time` is noisy if we need to debug.

Reply:
Record as a pointer in the task struct and return the pointer in getter will cause some data race.  It looks like the time.Time{} won't be saved in Marshal, [example](https://play.golang.org/p/ZeJ0RyyBrk)